File: masm-helper/src/extension.ts
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from "vscode";
import { MasmTaskProvider } from "./MasmTaskProvider";

// Constants
const RESOURCES_DIR: string = "resources" as const;

// Variables
let masmTaskProvider: vscode.Disposable | undefined;

// This method is called when your extension is activated
export function activate(context: vscode.ExtensionContext) {
	const resourcesPath = context.asAbsolutePath(RESOURCES_DIR);

	masmTaskProvider = vscode.tasks.registerTaskProvider(
		MasmTaskProvider.TaskType,
		new MasmTaskProvider(
			context.extension.packageJSON["name"],
			resourcesPath
		)
	);
}

// This method is called when your extension is deactivated
export function deactivate() {
	if (masmTaskProvider) {
		masmTaskProvider.dispose();
	}
}

File: masm-helper-kt/extension/src/commonMain/kotlin/Extension.kt
const val EXTENSION_NAME: String = "masm-helper"
const val RESOURCES_DIR: String = "./kotlin"

var masmTaskProvider: vscode.Disposable? = null

// This is the main entry point of the application.
@JsExport
@JsName("activate")
fun activate(context: vscode.ExtensionContext) {
    console.log("activate()")
    vscode.window.showInformationMessage("Hello World from Kotlin/JS!", "")

    val resourcesPath = context.asAbsolutePath(RESOURCES_DIR)

    masmTaskProvider = vscode.tasks.registerTaskProvider(
        MasmTaskProvider.TASK_TYPE,
        MasmTaskProvider(EXTENSION_NAME, resourcesPath)
    )
}

@JsExport
@JsName("deactivate")
fun deactivate() {
    masmTaskProvider?.dispose()
}